plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -562,8 +564,16 @@ public void error(StorageException exception) {
                             }
                         }
                     });
+                    pendingDeletesInBatch++;
+                    if (pendingDeletesInBatch % MAX_DELETES_PER_BATCH == 0) {
```

Comment:
I think the `%` here is unnecessary, we can just check for >=, since we reset when submitting the batch?